### PR TITLE
Dependency maintenance

### PR DESCRIPTION
**Pull request by johnduhart**
- Updates several project dependencies
- Introduces GitHub Actions for CI on Mac and Windows

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        run: poetry install --no-interaction
+      - run: poetry run flake8
+
+  test:
+    name: ${{ matrix.os }} / ${{ matrix.python-version }}
+    runs-on: ${{ matrix.image }}
+    strategy:
+      matrix:
+        os: [Ubuntu, macOS, Windows]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```
Comment (ykitamura-mdsol) [inline .github/workflows/check.yaml:30]: oh, we need to add `3.11` but maybe I can work on it later

File: pyproject.toml [hunk 2]
```diff
@@ -22,11 +22,11 @@ classifiers = [
...
 
 [tool.poetry.dependencies]
-python = "^3.7.13"
+python = "^3.7"
 requests = "^2.23"
-cachetools = "^4.1"
+cachetools = "^5.3"
 rsa = "^4.0"
-importlib-metadata = {version = "^3.6", python = "<3.8"}
+importlib-metadata = "^6.4"
```
Comment (ykitamura-mdsol) [inline pyproject.toml:29]: Is there any reason for removing `python = "<3.8"`?

Comment (ykitamura-mdsol) [reply pyproject.toml:29]: oh, I see. It makes sense.
To use `importlib-metadata` in all versions, I guess we need to remove these lines:

https://github.com/mdsol/mauth-client-python/blob/f08280d521c4b1bbf270d58200f342daa2cc8777/mauth_client/__init__.py#L2-L5

Comment (johnduhart) [reply pyproject.toml:29]: To be fair I don't understand what the library does or why it's needed. Potentially we can get away with loosing the version specification to set a minimum version if we don't need features provided in newer versions.

Comment (ykitamura-mdsol) [reply pyproject.toml:29]: I don't think we need new features. This dependency can be removed when we drop python `3.7` support.

**Review by ykitamura-mdsol — APPROVED** [review]

**Review by ejinotti-mdsol — APPROVED** [review]

**Pull request merged**